crates/core/src/routing_info/mod.rs

```diff
@@ -0,0 +1,219 @@
+
+        let update_interval = Configuration::pinned()
+            .common
+            .metadata_update_interval
+            .into();
+        let mut update_interval = tokio::time::interval(update_interval);
+        update_interval.set_missed_tick_behavior(MissedTickBehavior::Delay);
+        let mut cancel = pin!(cancellation_watcher());
+
+        loop {
+            tokio::select! {
+                _ = &mut cancel => {
+                    info!("Routing information refresher stopped");
+                    break;
+                }
+                Some(cmd) = self.receiver.recv() => {
+                    match cmd {
+                        Command::SyncRoutingInformation(target_version) => {
+                            self.spawn_sync_routing_information_task(target_version);
+                        }
+                    }
+                }
+                _ = update_interval.tick() => {
+                    debug!("Refreshing routing information...");
+                    self.spawn_sync_routing_information_task(TargetVersion::Latest);
+                }
+            }
+        }
+        Ok(())
+    }
+
+    fn spawn_sync_routing_information_task(&mut self, target_version: TargetVersion) {
+        if !self
+            .inflight_refresh_task
+            .as_ref()
+            .is_some_and(|t| !t.is_finished())
+        {
+            let partition_to_node_mappings = self.inner.clone();
+            let metadata_store_client = self.metadata_store_client.clone();
+
+            let task = task_center().spawn_unmanaged(
+                crate::TaskKind::Disposable,
+                "refresh-routing-information",
+                None,
+                {
+                    async move {
+                        sync_routing_information(
+                            partition_to_node_mappings,
+                            metadata_store_client,
+                            target_version,
+                            None,
+                        )
+                        .await
+                    }
+                },
+            );
+            self.inflight_refresh_task = task.ok();
+        };
+    }
+}
+
+pub fn spawn_partition_routing_refresher(
+    tc: &TaskCenter,
+    partition_routing_refresher: PartitionRoutingRefresher,
+) -> Result<TaskId, ShutdownError> {
+    tc.spawn(
+        TaskKind::MetadataBackgroundSync,
+        "partition-routing-refresher",
+        None,
+        partition_routing_refresher.run(),
+    )
+}
+
+async fn sync_routing_information(
+    partition_to_node_mappings: Arc<ArcSwap<PartitionToNodesRoutingTable>>,
+    metadata_store_client: MetadataStoreClient,
+    _target_version: TargetVersion,
+    sender: Option<oneshot::Sender<anyhow::Result<()>>>,
+) -> anyhow::Result<()> {
+    let scheduling_plan: SchedulingPlan = metadata_store_client
+        .get(SCHEDULING_PLAN_KEY.clone())
+        .await?
+        .context("Scheduling plan not found")?;
```

Comment:
What if there is no scheduler yet that has created a `SchedulingPlan`?

Reply:
I figure we'll just fail some requests until we have one - if a caller is trying to look up routing info for a partition, and we don't yet know how to address any of its nodes, it should return an appropriate error. E.g. the ingress layer might just return a 500 server-internal retryable error indicating to the upstream caller that the condition should be transient.

Reply:
I will add a debug-level log message about this at least.